client/my-sites/post-type-filter/index.jsx

```diff
@@ -3,44 +3,41 @@
  */
 import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
-import { find, includes, reduce } from 'lodash';
+import { compact, find, includes, reduce } from 'lodash';
 
 /**
  * Internal dependencies
  */
+import { areAllSitesSingleUser } from 'state/selectors';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { getSiteSlug, isJetpackSite } from 'state/sites/selectors';
-import { getNormalizedPostCounts } from 'state/posts/counts/selectors';
+import { isJetpackSite, isSingleUserSite, getSiteSlug } from 'state/sites/selectors';
+import { getNormalizedMyPostCounts, getNormalizedPostCounts } from 'state/posts/counts/selectors';
 import { mapPostStatus } from 'lib/route/path';
 import UrlSearch from 'lib/mixins/url-search';
 import QueryPostCounts from 'components/data/query-post-counts';
 import SectionNav from 'components/section-nav';
 import NavTabs from 'components/section-nav/tabs';
 import NavItem from 'components/section-nav/item';
 import Search from 'components/search';
+import AuthorSegmented from './author-segmented';
 
 const PostTypeFilter = React.createClass( {
 	mixins: [ UrlSearch ],
 
 	propTypes: {
+		authorToggleHidden: PropTypes.bool,
 		siteId: PropTypes.number,
-		query: PropTypes.object,
+		query: PropTypes.object.isRequired, // We need at least query.type
 		jetpack: PropTypes.bool,
 		siteSlug: PropTypes.string,
 		counts: PropTypes.object
 	},
 
-	getDefaultProps() {
-		return {
-			query: {}
-		};
-	},
-
 	getNavItems() {
-		const { query, siteSlug, jetpack, counts } = this.props;
+		const { query, siteId, siteSlug, jetpack, counts } = this.props;
 
 		return reduce( counts, ( memo, count, status ) => {
-			if ( ! jetpack && ! count && ! includes( [ 'publish', 'draft' ], status ) ) {
+			if ( siteId && ! ( jetpack && query.type !== 'post' ) && ! count && ! includes( [ 'publish', 'draft' ], status ) ) {
```

Comment:
Don't add the current status tab if:
* We're in single site mode and
* this is a WP.com site (not JP -- unless we're dealing with actual posts, not CPTs) and
* we actually have posts/CPTs with this status and
* it's neither 'publish' not 'draft' status

In other words,
* always add 'publish' and 'draft' tabs
* add all tabs in all-sites mode
* add all tabs in JP mode, for CPTs
* in all other cases, add status tabs only if there's at least one post/CPT with that status

I hope I got that right 😅 

I agree this is very convoluted -- but I was aiming for mimicking the current behavior as closely as possible (except for the [one inconsistency](https://github.com/Automattic/wp-calypso/pull/18113#issuecomment-332189245) that you noticed, but the previous behavior just seemed totally inconsistent, and reproducing it would've required a totally counter-intuitive criterion -- it felt plainly wrong).

Reply:
Added explanatory comment per 6fb2930.

Reply:
> Added explanatory comment

❤️